rclcpp/src/rclcpp/time.cpp

```diff
@@ -130,6 +140,14 @@ Time::operator builtin_interfaces::msg::Time() const
 }
 
 void
+Time::operator=(const Time & rhs)
+{
+  rcl_time_source_ = rhs.rcl_time_source_;
```

Comment:
Can you try that with a `RCL_ROS_TIME`? Since it internally allocates `data` within the `rcl_time_source_t` I would expect that this can't be just copied here.